pkg/handler/common/provider/kubevirt.go

```diff
@@ -224,3 +228,40 @@ func KubeVirtStorageClassesWithClusterCredentialsEndpoint(ctx context.Context, u
 
 	return KubeVirtStorageClasses(ctx, kvKubeconfig)
 }
+
+func filterVMIPresets(vmiPresets apiv2.VirtualMachineInstancePresetList, quota kubermaticv1.MachineDeploymentVMResourceQuota) apiv2.VirtualMachineInstancePresetList {
+	filteredVMIPresets := apiv2.VirtualMachineInstancePresetList{}
+
+	// Range over the records and apply all the filters to each record.
+	// If the record passes all the filters, add it to the final slice.
+	for _, vmiPreset := range vmiPresets {
+		presetSpec := kubevirtv1.VirtualMachineInstancePresetSpec{}
+		if err := json.Unmarshal([]byte(vmiPreset.Spec), &presetSpec); err != nil {
+			return filteredVMIPresets
+		}
+
+		var cpu int
+		if presetSpec.Domain.CPU != nil && presetSpec.Domain.CPU.Cores != 0 {
+			cores := presetSpec.Domain.CPU.Cores
+			// if threads and sockets are set, calculate VCPU
+			threads := presetSpec.Domain.CPU.Threads
+			if threads == 0 {
+				threads = 1
+			}
+			sockets := presetSpec.Domain.CPU.Sockets
+			if sockets == 0 {
+				sockets = 1
+			}
+			cpu = int(cores * threads * sockets)
+		}
+		if !presetSpec.Domain.Resources.Requests.Cpu().IsZero() {
+			cpu = int(presetSpec.Domain.Resources.Requests.Cpu().AsApproximateFloat64())
+		}
+
+		memory := int(presetSpec.Domain.Resources.Requests.Memory().Value() / (1 << 30))
+		if handlercommon.FilterCPU(cpu, quota.MinCPU, quota.MaxCPU) && handlercommon.FilterMemory(memory, quota.MinRAM, quota.MaxRAM) {
+			filteredVMIPresets = append(filteredVMIPresets, vmiPreset)
+		}
+	}
```

Comment:
I think this is still not fully correct. 
The CPU can also be set just in `presetSpec.Domain.Resources.Limits`, and also you should have a default value set (from what I understand its 100mi). 
And for the memory, it can also be just set in `presetSpec.Domain.Resources.Limits`, and the default is 1Gi